components/layout_2020/display_list/stacking_context.rs

```diff
@@ -1293,6 +1292,25 @@ impl BoxFragment {
                 StackingContextBuildMode::SkipHoisted,
             );
         }
+
+        if matches!(&fragment, Fragment::Box(box_fragment) if matches!(
+            box_fragment.borrow().specific_layout_info,
+            Some(SpecificLayoutInfo::TableGridWithCollapsedBorders(_))
+        )) {
+            stacking_context
+                .contents
+                .push(StackingContextContent::Fragment {
+                    scroll_node_id: new_scroll_node_id,
+                    reference_frame_scroll_node_id: reference_frame_scroll_node_id_for_fragments,
+                    clip_chain_id: new_clip_chain_id,
+                    // TODO: should this use `self.get_stacking_context_section()`?
+                    section: StackingContextSection::DescendantBackgroundsAndBorders,
```

Comment:
I guess you could check if this is correct by determining if a table that establishes a stacking context has border that behave properly when elements inside the table also form stacking contexts with `z-index` as well.

Reply:
So after thinking more, we are painting the borders with the grid. The grid doesn't inherit `position`, `float`, etc. so `get_stacking_context_section()` should be `StackingContextSection::DescendantBackgroundsAndBorders` anyways. It may actually be `StackingContextSection::OwnBackgroundsAndBorders`  because it inherits `display`, but this seems like a bug, I can address it in a follow-up. I guess it's better to land this as is, which makes us match Blink and WebKit. It seems Firefox paints the collapsed borders with the table wrapper instead of the table grid, that would explain the difference in behavior.